File: ios/Packages/FeatureWallet/Sources/FeatureWallet/Module/WalletModule.swift
import Contracts
import WireCore

/// Wallet feature module — accessible to ADMIN, STAFF, and CUSTOMER.
public final class WalletModule: AppModule {
    public init() {}

    public let metadata = ModuleMetadata(
        id: "wallet",
        name: "Wallet",
        requiredRoles: [.ADMIN, .STAFF, .CUSTOMER],
        priority: 80
    )

    public func initialize(context: any ModuleContext) {
        for slot in provideWidgets() {
            context.slotRegistry.register(slot)
        }
    }

    public func provideRoutes() -> [ModuleRoute] {
        [ModuleRoute(route: "/wallet", requiredRole: .CUSTOMER)]
    }
}

File: flutter/feature_wallet/lib/src/module/wallet_module.dart
import 'package:contracts/contracts.dart';

/// Wallet feature module — accessible to ADMIN, STAFF, and CUSTOMER.
class WalletModule extends AppModule {
  @override
  ModuleMetadata get metadata => const ModuleMetadata(
        id: 'wallet',
        name: 'Wallet',
        requiredRoles: {Role.ADMIN, Role.STAFF, Role.CUSTOMER},
        priority: 80,
      );

  @override
  void initialize(ModuleContext context) {
    for (final slot in provideWidgets()) {
      context.slotRegistry.register(slot);
    }
  }

  @override
  List<ModuleRoute> provideRoutes() => [
        const ModuleRoute(route: '/wallet', requiredRole: Role.CUSTOMER),
      ];
}

